### PR TITLE
DWR-374, DWR-375 and some other fixes

### PR DESCRIPTION
**Pull request by malaffoon**
comments inline

### CODE REVIEW

File: data_generator/config/cfg.py [hunk 1]
```diff
@@ -49,34 +49,40 @@
 ASMT_STATUS_INACTIVE = 'I'
 ASMT_STATUS_DELETED = 'D'
 
-ASMT_SCORE_MIN = 1200
```
Comment (malaffoon) [inline data_generator/config/cfg.py]: Replace these silly values with numbers from SB documentation.

File: data_generator/config/out.py [hunk 2]
```diff
@@ -302,7 +302,6 @@
                                {'name': 'asmt_cut_point_1', 'val': 'assessment.overall_cut_point_1'},
                                {'name': 'asmt_cut_point_2', 'val': 'assessment.overall_cut_point_2'},
                                {'name': 'asmt_cut_point_3', 'val': 'assessment.overall_cut_point_3'},
-                               {'name': 'asmt_cut_point_4', 'val': 'assessment.overall_cut_point_4'},
```
Comment (malaffoon) [inline data_generator/config/out.py]: Correct me if i'm wrong, but there is never ever a 4th cut-point, right?

File: data_generator/generators/iab_assessment.py [hunk 3]
```diff
@@ -67,6 +68,7 @@ def generate_interim_assessment(asmt_year: int,
         raise KeyError("Subject '%s' not found in claim definitions" % subject)
 
     claims = claim_definitions[subject]
+    asmt_scale_scores = sbac_config.ASMT_SCALE_SCORE[subject][grade]
```
Comment (malaffoon) [inline data_generator/generators/iab_assessment.py]: Look up the LOSS, HOSS and cutpoints based on subject and grade, per SB specs.

File: data_generator/generators/iab_assessment.py [hunk 4]
```diff
@@ -155,42 +155,27 @@ def generate_interim_assessment_outcome(date_taken: datetime.date,
     # set timestamps for the opportunity
     gen_asmt_generator.set_opportunity_dates(sao)
 
-    demographics = sbac_config.DEMOGRAPHICS_BY_GRADE[student.grade]
```
Comment (malaffoon) [inline data_generator/generators/iab_assessment.py]: moved all this into a helper

File: data_generator/model/assessment.py [hunk 5]
```diff
@@ -21,8 +21,7 @@ class Assessment:
                  'claim_4_score_max',
                  'claim_1_score_weight', 'claim_2_score_weight', 'claim_3_score_weight', 'claim_4_score_weight',
                  'claim_perf_lvl_name_1', 'claim_perf_lvl_name_2', 'claim_perf_lvl_name_3',
-                 'overall_cut_point_1', 'overall_cut_point_2', 'overall_cut_point_3', 'overall_cut_point_4',
-                 'claim_cut_point_1', 'claim_cut_point_2',
```
Comment (malaffoon) [inline data_generator/model/assessment.py]: There's really no such thing as "claim" cut-points. And they weren't used anywhere anyway.

File: data_generator/outputworkers/xml_worker.py [hunk 6]
```diff
@@ -112,7 +116,7 @@ def write_asmt_to_file(self, outcome: AssessmentOutcome):
         opportunity.set('sessionId', outcome.session)
         opportunity.set('windowId', 'WINDOW_ID')    # TODO
         # opportunity.set('windowOpportunity', None)
-        opportunity.set('administrationCondition', 'Valid')
+        opportunity.set('administrationCondition', outcome.admin_condition)
```
Comment (malaffoon) [inline data_generator/outputworkers/xml_worker.py]: DWR-375

File: data_generator/outputworkers/xml_worker.py [hunk 7]
```diff
@@ -123,8 +127,17 @@ def write_asmt_to_file(self, outcome: AssessmentOutcome):
             segment.set('algorithm', outcome.assessment.segment.algorithm)
             segment.set('algorithmVersion', outcome.assessment.segment.algorithm_version)
 
-        self.add_overall_score(opportunity, 'ScaleScore', outcome.overall_score, (outcome.overall_score - outcome.overall_score_range_min))
-        self.add_overall_score(opportunity, 'PerformanceLevel', outcome.overall_perf_lvl)
+        self.add_scale_score(opportunity, 'Overall',
```
Comment (malaffoon) [inline data_generator/outputworkers/xml_worker.py]: Emit claim scores for ICA/SUM.

File: data_generator/readers/tabulator_reader.py [hunk 8]
```diff
@@ -82,12 +82,14 @@ def __load_row(row, asmt: Assessment, parse_asmt, parse_item):
...
         asmt.year = int(row['AcademicYear']) + 1  # TODO - remove +1 when tabulator is fixed
         asmt.bank_key = row['BankKey']
-        asmt.overall_score_min = __getScore(row['ScaledLow1'])
-        asmt.overall_cut_point_1 = __getScore(row['ScaledHigh1'])
-        asmt.overall_cut_point_2 = __getScore(row['ScaledHigh2'])
-        asmt.overall_cut_point_3 = __getScore(row['ScaledHigh3'])
-        asmt.overall_cut_point_4 = None
-        asmt.overall_score_max = __getScore(row['ScaledHigh4'])
+
+        asmt_scale_scores = cfg.ASMT_SCALE_SCORE[asmt.subject][asmt.grade]
```
Comment (malaffoon) [inline data_generator/readers/tabulator_reader.py]: Tabulator has a bug where the scale scores aren't included for ICA's ... this uses them if present but falls back to hardcoded defaults.

File: data_generator/util/assessment_stats.py [hunk 9]
```diff
@@ -216,3 +216,29 @@ def random_claims(score: int, claim_weights: [float], claim_min: int, claim_max:
...
+def random_claim_error(claim_score: int, claim_min: int, claim_max: int):
+    """Generate a std error for a claim score.
+    Not sure if it is valid but this will give a larger error, the lower the score.
+
+    :param claim_score: score
+    :param claim_min: min possible score
+    :param claim_max: max possible score
+    :return: std error
+    """
+    return 40 + random.randint(0, round(80 * (claim_max - claim_score) / (claim_max - claim_min)))
```
Comment (malaffoon) [inline data_generator/util/assessment_stats.py]: 40-120 seems to be a reasonable std error; error band will be 80-240.

File: data_generator/worker_manager.py [hunk 10]
```diff
@@ -81,6 +81,10 @@ def run(self):
 
         assessments = self.__assessment_packages()
 
+        if len(assessments) == 0:
```
Comment (malaffoon) [inline data_generator/worker_manager.py]: unrelated, just noticed

File: data_generator/model/assessment.py (hunk 5 above)
Comment (agorina) [inline data_generator/model/assessment.py]: That is correct. Claims are scored as IABs, and hence they have cut points that are specific to each student. That is my understanding.

File: data_generator/outputworkers/xml_worker.py (hunk 7 above)
Comment (agorina) [inline data_generator/outputworkers/xml_worker.py]: Yeah!

**Pull request merged**